### PR TITLE
[Icon] Add dynamic source connect icon

### PR DESCRIPTION
**Pull request by Rusty-UX**
datasource-connect

### Adds support for dynamic source connect icon
* This icon is used when a dynamic source is connected, and powering part of the experience: either data, or functionality.
* This icon is used in context of Shopify Admin, and Shopify Mobile for merchandising flows.

### WHY are these changes introduced?
* Provides additional specificity in how we communicate dynamic source connections in admin and online store

### WHAT is this pull request doing?
* Adds support for dynamic source connect icon
* Updates description for add dynamic source icon

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by Rusty-UX** [comment]
@kyledurand Updated with a new PR, I think my previous one might've just been out of date 🤞 

Note: I've added a changeset as well - can you confirm that the right versioning has been added?

File: .changeset/six-pumpkins-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': major
```
Comment (kyledurand) [inline .changeset/six-pumpkins-mix.md]:
```suggestion
'@shopify/polaris-icons': minor
```

Comment (Rusty-UX) [reply .changeset/six-pumpkins-mix.md]: Thanks bub

**Review by kyledurand — APPROVED** [review]
👍

File: .changeset/six-pumpkins-mix.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': minor
+---
+
+Adds support for connect dynamic source iconn
```
Comment (kyledurand) [inline .changeset/six-pumpkins-mix.md]: Just noticed this typo (iconn). Let's maybe just say:

```suggestion
Added DatabaseConnectIcon
```

**Review by heyjoethomas — APPROVED** [review]

**Pull request merged**